staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -284,23 +302,61 @@ func (a *GenericAdmissionWebhook) callHook(ctx context.Context, h *v1alpha1.Exte
 }
 
 func (a *GenericAdmissionWebhook) hookClient(h *v1alpha1.ExternalAdmissionHook) (*rest.RESTClient, error) {
-	serverName := h.ClientConfig.Service.Name + "." + h.ClientConfig.Service.Namespace + ".svc"
-	u, err := a.serviceResolver.ResolveEndpoint(h.ClientConfig.Service.Namespace, h.ClientConfig.Service.Name)
-	if err != nil {
-		return nil, err
+	if client, ok := a.cache.get(&h.ClientConfig); ok {
+		return client, nil
 	}
 
-	// TODO: cache these instead of constructing one each time
```

Comment:
@deads2k I think what we want to cache is the client itself, rather than the config?

Reply:
Done

Reply:
Could you change the PR title?

Reply:
done